### ISSUE TITLE
Uber JAR for Spline

### ISSUE THREAD

**Issue by wajda** [AbsaOSS/spline#163]
Requested by Arvind:
> regarding the fat jar, it would be VERY useful if your team builds it and publishes to Maven. The reason for this is because in Azure the most popular Spark deployment is Azure Databricks. Now, Databricks (in Azure and in AWS, but I am only looking at Azure) users are predominantly coding via. notebooks as opposed to submitting a traditional JAR to the cluster. Given this notebook interface and the user profiles for our customers, we have seen that if there are too many dependency JARs have to be added as libraries, it causes adoption friction and the customer would most likely not use the library. Hence, if you can release an uber JAR officially on Maven that would be really helpful to drive adoption of Spline with Azure Databricks.

**Comment by wajda** [comment]
@AbsaOSS/spline

**Comment by algattik** [comment]
In addition/in parallel, it would be extremely useful to generate different jars / different uber-jars for the various Spark versions, rather than have transitive dependency resolution via maven profiles.
Other projects routinely do this, for example
https://search.maven.org/artifact/com.microsoft.azure/azure-cosmosdb-spark_2.3.0_2.11/1.3.3/jar

This would greatly simplify the installation e.g. on Databricks:
<img width="582" alt="image" src="https://user-images.githubusercontent.com/22099722/55028540-45a65580-5008-11e9-82da-1a1eb4865a16.png">

**Comment by jcnorman48** [comment]
+1 I think this would resolve the issue as well https://github.com/AbsaOSS/spline/issues/160
I see the same ${spark.version} error in DataBricks and SBT

**Comment by wajda** [comment]
@algattik , @jcnorman48 
The Uber-jar is available in Maven central. Can you please test it give us feedback? Thank you.
https://search.maven.org/search?q=spline-bundle

**Comment by algattik** [comment]
Running on Databricks:
java.lang.NoClassDefFoundError: scalaz/Semigroup
	at za.co.absa.spline.core.harvester.DataLineageBuilder$$anonfun$getMetrics$1.apply(DataLineageBuilder.scala:143

I think you might be excluding too much. When I built my uber JAR I used:

```
<dependencies>
    <dependency>
      <groupId>za.co.absa.spline</groupId>
      <artifactId>spline-core</artifactId>
      <version>${spline.version}</version>
    </dependency>
    <dependency>
      <groupId>za.co.absa.spline</groupId>
      <artifactId>spline-core-spark-adapter-${spark.version}</artifactId>
      <version>${spline.version}</version>
      <exclusions>
        <exclusion>
          <groupId>org.apache.spark</groupId>
          <artifactId>*</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
    <dependency>
      <groupId>za.co.absa.spline</groupId>
      <artifactId>spline-persistence-mongo</artifactId>
      <version>${spline.version}</version>
    </dependency>
  </dependencies>
```

**Comment by wajda** [comment]
yep, got it fixed already :) 9771b7464f3c55b5bb03a5983a24b31fa8188e24
Sorry, did have a chance to test it properly last time. Will release 0.3.8 asap.
